presto-main/src/main/java/com/facebook/presto/operator/PagesSpatialIndexSupplier.java

```diff
@@ -81,8 +84,10 @@ public PagesSpatialIndexSupplier(
             Optional<Integer> partitionChannel,
             SpatialPredicate spatialRelationshipTest,
             Optional<JoinFilterFunctionCompiler.JoinFilterFunctionFactory> filterFunctionFactory,
-            Map<Integer, Rectangle> partitions)
+            Map<Integer, Rectangle> partitions,
+            LocalMemoryContext localUserMemoryContext)
     {
+        requireNonNull(localUserMemoryContext);
```

Comment:
add error message: `requireNonNull(localUserMemoryContext, "localUserMemoryContext is null");`